idem_azurerm/states/azurerm/network/network_profile.py

```diff
@@ -0,0 +1,300 @@
+    authentication parameters are case sensitive.
+
+    Required provider parameters:
+
+    if using username and password:
+      * ``subscription_id``
+      * ``username``
+      * ``password``
+
+    if using a service principal:
+      * ``subscription_id``
+      * ``tenant``
+      * ``client_id``
+      * ``secret``
+
+    Optional provider parameters:
+
+    **cloud_environment**: Used to point the cloud driver to different API endpoints, such as Azure GovCloud. Possible values:
+      * ``AZURE_PUBLIC_CLOUD`` (default)
+      * ``AZURE_CHINA_CLOUD``
+      * ``AZURE_US_GOV_CLOUD``
+      * ``AZURE_GERMAN_CLOUD``
+
+    Example configuration for Azure Resource Manager authentication:
+
+    .. code-block:: yaml
+
+        azurerm:
+            default:
+                subscription_id: 3287abc8-f98a-c678-3bde-326766fd3617
+                tenant: ABCDEFAB-1234-ABCD-1234-ABCDEFABCDEF
+                client_id: ABCDEFAB-1234-ABCD-1234-ABCDEFABCDEF
+                secret: XXXXXXXXXXXXXXXXXXXXXXXX
+                cloud_environment: AZURE_PUBLIC_CLOUD
+            user_pass_auth:
+                subscription_id: 3287abc8-f98a-c678-3bde-326766fd3617
+                username: fletch
+                password: 123pass
+
+    The authentication parameters can also be passed as a dictionary of keyword arguments to the ``connection_auth``
+    parameter of each state, but this is not preferred and could be deprecated in the future.
+
+    Example states using Azure Resource Manager authentication:
+
+    .. code-block:: jinja
+
+        Ensure network profile exists:
+            azurerm.network.network_profile.present:
+                - name: aci-network-profile
+                - resource_group: testgroup
+                - container_network_interface_configurations:
+                    - name: eth0
+                      ip_configurations:
+                        - name: ipconfigprofile
+                          subnet:
+                              id: {{ subnet_resource_id }}
+                - tags:
+                    how_awesome: very
+                    contact_name: Elmer Fudd Gantry
+
+        Ensure network profile is absent:
+            azurerm.network.network_profile.absent:
+                - name: aci-network-profile
+                - resource_group: testgroup
+
+"""
+# Import Python libs
+from dict_tools import differ
+import logging
+
+
+log = logging.getLogger(__name__)
+
+
+async def present(
+    hub,
+    ctx,
+    name,
+    resource_group,
+    container_network_interfaces=None,
+    container_network_interface_configurations=None,
+    tags=None,
+    connection_auth=None,
+    **kwargs,
+):
+    """
+    .. versionadded:: 3.0.0
+
+    Ensure a network profile exists.
+
+    :param name: The name of the network profile.
+
+    :param resource_group: The name of the resource group to which the network profile belongs.
+
+    :param container_network_interfaces: List of child container network interfaces.
```

Comment:
Should we add more information regarding what specifically needs to be passed (i.e., a dictionary with the key value pairs 1, 2, and 3, etc.)? This is more of a suggestion to help avoid a user getting confused or not realized what parameters are fully required.

Reply:
they are really complex structures with nested objects... and I don't see people using this state very much. I'll link the SDK docs to give direction and I'll cover it in my "how to containers" topic for the docs.